sbp.bash

```diff
@@ -39,10 +56,9 @@ _sbp_set_prompt() {
     title="${HOSTNAME:-ssh}:${title}"
   fi
   printf '\e]2;%s\007' "$title"
-
-  PS1=$(bash "${SBP_PATH}/src/main.bash" "$command_status" "$command_duration")
+  # gitstatus.plugin.sh requires an interactive shell
+  PS1=$(bash --noediting --noprofile --norc -i "${SBP_PATH}/src/main.bash" "$command_status" "$command_duration" "$GITSTATUS_DIR" "$GITSTATUS_DAEMON_PID" "$_GITSTATUS_REQ_FD" "$_GITSTATUS_RESP_FD")
```

Comment:
Did you verify that this is the case? I can't see the rationale for this looking at the repository at least as the gitstatus binary is doing the actual work.

Reply:
If you look at the [script](https://github.com/romkatv/gitstatus/blob/master/gitstatus.plugin.sh#L3) you can see the same return that the default ``.bashrc`` has.

If we would vendor the script we could remove the check and remove this giant workaround. Not sure what to do.

Reply:
ah right. Would we be able to limit the interactive shell to just the invocation of the gitplugin that we are doing in the segment script though? I'm thinking that this could probably be a  separate segment script, as it's using a completely different approach than the regular git segment. Does that make sense to you?

Reply:
Not sure. We can't easily forward all variables we need without explicitly listing all of them.

https://github.com/brujoand/sbp/blob/f5d0b29e2647ce644ea01be469da10a3cbb4939d/src/execute.bash#L46 the segments are loaded here as far as I can tell and converting source to bash is not that easy.

I wouldn't make this a different segment because it is still sharing some of the code.